app/views/lookup_keys/_value.html.erb

```diff
@@ -9,7 +9,7 @@
     <%= text_field_tag '', nil, :class => 'matcher_value' %>
   </td>
   <td>
-    <%= hidden_value_field(f, :value, f.object.value, f.object.use_puppet_default, :'data-property' => 'value') %>
+    <%= hidden_value_field(f, :value, f.object.value_before_type_cast, f.object.use_puppet_default, :'data-property' => 'value') %>
```

Comment:
This doesn't seem to work on the Puppet class edit form, as when entering an invalid value for a hash smart variable's matcher and submitting the form, it reloads with the `--- ! '...'` variant.  It looks like the `value` passed into the helper is only used when creating hidden value fields, not visible fields.